app/store.js

```diff
@@ -17,11 +18,37 @@ export const store = new Vuex.Store({
         );
       }
     },
-    save(state, data) {
-        /*state.name = data.name;
-        state.firstLoad = data.firstLoad;*/
-        Object.assign(state, data);
+    save(state) {
+        // Object.assign(state, data);
         ApplicationSettings.setString("store", JSON.stringify(state));
+    },
+    setName(state, name) {
+      state.name = name;
+    },
+    setGoal(state, goal) {
+      state.goal = goal;
+    },
+    setFirstLoad(state, firstLoad) {
+      state.firstLoad = firstLoad;
+    }
+  },
+
+  getters: {
+    //
+  },
+
+  actions: {
```

Comment:
Initially we called `save` directly on the `$store` which would save the state to `ApplicationSettings` directly. Now that we have refactored it to use actions, we should also save the settings after we change the state.